core/document/src/document.rs

```diff
@@ -227,13 +227,18 @@ impl Document {
 		Ok(())
 	}
 
-	pub fn layer_axis_aligned_bounding_box(&self, path: &[LayerId]) -> Result<[DVec2; 2], DocumentError> {
-		let layer = self.layer(path)?;
-		Ok(layer.bounding_box(self.root.transform * layer.transform, layer.style))
+	pub fn layer_axis_aligned_bounding_box(&self, path: &[LayerId]) -> Result<Option<[DVec2; 2]>, DocumentError> {
+		if let &[] = path {
+			// Special case for root. Root's local is the documents global, so we avoid transforming its transform by itself.
+			self.layer_local_bounding_box(path)
+		} else {
+			let layer = self.document_layer(path)?;
+			Ok(layer.bounding_box(self.root.transform * layer.transform, layer.style))
+		}
```

Comment:
This will somewhat be replaced with the new transform api, so could you maybe add a TODO: Replace with functions of the transform api
Just for future reference

Reply:
Done